lib/req/steps.ex

```diff
@@ -227,10 +227,10 @@ defmodule Req.Steps do
   def compressed(request) do
     case Map.fetch(request.options, :compressed) do
       :error ->
-        put_new_header(request, "accept-encoding", supported_accept_encoding())
+        Req.Request.put_header(request, "accept-encoding", supported_accept_encoding())
```

Comment:
there is an important difference between put_new_header and the new function. With the latter, it would be impossible for the user to set accept-encoding header themselves, it would always be overridden by the compressed step. I think that's undesirable.

We could add a Req.Request.put_new_header but let's hold off for more use cases. For now, please revert this and similar changes.

Reply:
Actually please add put_new_header but add doc: false